### PR TITLE
Doc fix for `DropdownButtonFormField.value`

### LINKED ISSUE #144135: DropDownButtonFormfield documentation references 'initialValue', but it doesn't exist.
When trying to reference `initialValue` for `DropDownButtomFormfield` as described in documentation

https://api.flutter.dev/flutter/material/DropdownButtonFormField-class.html

it does not exist.

image

Can you please update documentation to use `value` instead.

### PR DESCRIPTION
**Pull request by QuncCccccc**
Fixes https://github.com/flutter/flutter/issues/144135

This PR is just a doc fix.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by matanlurey** [comment]
This is not a doc fix, it changes the API of `DropdownButtonFormField`,

That is, a previous:

```dart
class CustomDropdownButtomFormField implements DropdownButtonFormField { ... }
```

... is now broken by this change.

In addition, it's confusing, because now you have `this.initialValue` (from `FormField.initialValue`) and `this.value` which are the same and both immutable. I don't think the initial request was valid.

**Review by matanlurey — CHANGES_REQUESTED** [review]
I don't think this makes sense, PTAL at my comment on the PR.

**Comment by QuncCccccc** [comment]
> I don't think this makes sense, PTAL at my comment on the PR.

Reverted the change and updated the doc for FormField and DropdownButtonFormField! Thanks for pointing this out!

File: packages/flutter/lib/src/material/dropdown.dart [hunk 1]
```diff
@@ -1555,6 +1555,8 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
 /// pass a [GlobalKey] to the constructor and use [GlobalKey.currentState] to
 /// save or reset the form field.
 ///
+/// The `value` parameter maps to [FormField.initialValue].
```
Comment (matanlurey) [inline packages/flutter/lib/src/material/dropdown.dart:1558]: You maybe can do `[value]`, but defer to y'all (I don't update user-facing docs often and don't know the considerations).

**Review by matanlurey — APPROVED** [review]

File: packages/flutter/lib/src/material/dropdown.dart (hunk 1 above)
Comment (QuncCccccc) [reply packages/flutter/lib/src/material/dropdown.dart:1558]: I see both [] and `` are used for constructor parameters. Let me double check!

Comment (QuncCccccc) [reply packages/flutter/lib/src/material/dropdown.dart:1558]: Should be backticks for this case as it is not a field in `DropdownButtonFormField`:) https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#dartdoc-specific-requirements

**Pull request merged**